ldk/csharp/OliveHelpsLDK/OliveHelpsLDK/Network/NetworkClient.cs

```diff
@@ -29,18 +32,43 @@ public Task<HTTPResponse> HTTPRequest(HTTPRequest request, CancellationToken can
                 Method = request.Method,
                 Url = request.URL
             };
+
+            message.Headers.Add(request.Headers);
+
             return Client.HTTPRequestAsync(message, CreateOptions(cancellationToken)).ResponseAsync
                 .ContinueWith(LoggedParser<Task<HTTPResponseMsg>, HTTPResponse>(task => FromProto(task.Result)),
                     cancellationToken);
         }
 
         internal static HTTPResponse FromProto(HTTPResponseMsg response)
         {
-            return new HTTPResponse
+            var clientResponse = new HTTPResponse
             {
                 Data = response.ToByteArray(),
                 ResponseCode = checked((int) response.ResponseCode)
             };
+
+            clientResponse.Headers = ParseHeaders(response.Headers);
+
+            return clientResponse;
+        }
+
+        private static IDictionary<string, IList<string>> ParseHeaders(MapField<string, ListValue> headers)
+        {
+            var parsedHeaders = new Dictionary<string, IList<string>>();
```

Comment:
I can do these as the `Linq` based variants that do `map`, `filter`, `reduce`, if desired. Those are debatably more idiomatic, though I doubt it matters for internal code.

Reply:
I agree - this is fine as-is